base/reinterpretarray.jl

```diff
@@ -475,6 +497,10 @@ end
     v = convert(T, v)::T
     # Make sure to match the scalar reinterpret if that is applicable
     if sizeof(T) == sizeof(S) && (fieldcount(T) + fieldcount(S)) == 0
+        if Base.issingletontype(T) # singleton types
+            @boundscheck checkbounds(a, i1, tailinds...)
+            return T.instance # setindex! is a noop here
```

Comment:
```suggestion
            return a # setindex! is a noop here
```

This function is supposed to return the argument, though looks like several of the cases are wrong.

Reply:
You're right, sorry about that! I fixed both instances in e63a8fb.
I also fixed the currently erroneous return value for `setindex!` on some `reinterpretarray`s where it would return the parent array:
```julia
julia> t = reinterpret(Int, UInt[15])
1-element reinterpret(Int64, ::Vector{UInt64}):
 15

julia> setindex!(t, 7, 1) # should return t instead of t.parent
1-element Vector{UInt64}:
 0x0000000000000007
```
That's a bugfix (I believe) but it's technically breaking so let me know if you prefer I'd move that to another PR.